### PR TITLE
Fix RouterTabs issue with child routes

### PR DESCRIPTION
**Pull request by johnnyomair**
This fix resolves an issue where the default route would not be rendered when the path of the URL would not match exactly (e.g. a child route). This is done by removing the exact prop from the Route component and using a Switch instead. The Route component with the default path "" is rendered last to ensure that another, more specific route (e.g. "/page2") is picked first by the Switch, using the default route as a fallback.

### CODE REVIEW

**Comment by rabengraph** [comment]
The pull-request looks good to me!
But I don't fully grasp the issue.
Does the wrong behaviour only occur with nested routes?
Would your storybook-example fail with the current implementation?
If yes, how would the failure look like?

**Comment by johnnyomair** [comment]
Well, the issue is hard to describe. Here's a GIF showing the added story with the current implementation.

navigation_fail

Once I activate the nested route using the `StackSwitchApi`, the `RouterTab` doesn't render the respective route, as the route doesn't match `exact`.
Here's a GIF showing the same story with the implemented fix:

navigation_success

Here, the respective child route is rendered correctly.

File: packages/react-admin-core/src/RouterTabs.tsx [hunk 1]
```diff
@@ -67,7 +77,7 @@ class RouterTabs extends React.Component<IProps> {
                                     <Route path={`${this.props.match.url}/:tab`}>
                                         {({ match }) => {
                                             const routePath = match ? "/" + match.params.tab : "";
-                                            const value = paths.indexOf(routePath);
+                                            const value = paths.includes(routePath) ? paths.indexOf(routePath) : defaultPathIndex;
```
Comment (rabengraph) [inline packages/react-admin-core/src/RouterTabs.tsx:80]: This is definitely a good idea, it prevents a value of '-1' (no index found)

File: packages/react-admin-core/src/RouterTabs.tsx [hunk 2]
```diff
@@ -105,30 +115,32 @@ class RouterTabs extends React.Component<IProps> {
...
-                                        </StackBreadcrumb>
-                                    );
-                                } else {
-                                    return <TabContainer style={{ display: "none" }}>{child.props.children}</TabContainer>;
-                                }
-                            }}
-                        </Route>
-                    ) : null;
-                })}
+                <Switch>
```
Comment (rabengraph) [inline packages/react-admin-core/src/RouterTabs.tsx:118]: Switch makes sense here in my opinion!

File: packages/react-admin-core/src/RouterTabs.tsx [hunk 3]
```diff
@@ -57,6 +57,16 @@ class RouterTabs extends React.Component<IProps> {
             return child.props.path;
         });
 
+        const rearrangedChildren = React.Children.toArray(this.props.children);
+        const defaultPathIndex = paths.indexOf("");
+
+        if (defaultPathIndex >= 0) {
+            const defaultPathChild = rearrangedChildren[defaultPathIndex];
+
+            rearrangedChildren.splice(defaultPathIndex, 1);
```
Comment (rabengraph) [inline packages/react-admin-core/src/RouterTabs.tsx:66]: the storybook-example works correctly even without rearranging the children,
can we have a story where re-arranging the children matters ?

Comment (johnnyomair) [reply packages/react-admin-core/src/RouterTabs.tsx:66]: Navigation to /page2 (Tab "Page 2") doesn't work without rearranging the children.

**Review by nsams — APPROVED** [review]

**Comment by nsams** [comment]
warum failed der build?

**Comment by nsams** [comment]
> warum failed der build?

versuche hier zu fixen: #221

**Review by rabengraph — APPROVED** [review]

**Pull request merged**